app/src/features/surveys/view/SurveyHeader.tsx

```diff
@@ -238,8 +276,15 @@ const SurveyHeader: React.FC<ISurveyHeaderProps> = (props) => {
                 </Typography>
               </Box>
             </Box>
-            <Box ml={4} mb={4}>
-              {showDeleteSurveyButton && (
+            <Box ml={4}>
+              {showServerAdminButtons && (
+                <Button size="small" color="primary" variant="outlined" onClick={showUploadSurveyDialog}>
+                  Submit Data
+                </Button>
+              )}
+            </Box>
```

Comment:
You could replace the `showServerAdminButtons` with just a `SystemRoleGuard` wrapper.  Examples in `Header.tsx`